### PR TITLE
feat/fmd-206 add ingestion timing

### PR DESCRIPTION
**Pull request by tom-webber**
- **feat: add calls to `time` unix command to `datahub ingest` calls**
- **fix: Notify on failure GHA steps referencing incorrect `input` var for environment**
- **chore: type checker linting (fixes pylance type checking errors)**
- **feat: add st_time decorator for timing function runs**
- **chore: use non-depricated suggested methods from datahub utils**

### CODE REVIEW

File: ingestion/transformers/assign_cadet_databases.py [hunk 1]
```diff
@@ -14,9 +14,13 @@
...
 from ingestion.dbt_manifest_utils import (
     get_cadet_manifest,
-    validate_fqn,
     parse_database_and_table_names,
+    validate_fqn,
 )
+from ingestion.utils import st_time
+
+logging.basicConfig()
+logging.getLogger(__name__).setLevel(logging.DEBUG)
```
Comment (MatMoore) [inline ingestion/transformers/assign_cadet_databases.py:23]: What is this `setLevel` for? It looks like nothing is logged to this logger at the moment, as the return value of `logging.getLogger` is discarded. Was the intention to set a `logger` variable and use it? Or get the root logger?

File: ingestion/utils.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import time
+from datetime import timedelta
+
+logging.basicConfig()
+logging.getLogger(__name__).setLevel(logging.DEBUG)
+
+
+def st_time(func):
+    """
+    st decorator to calculate the total time of a func
```
Comment (MatMoore) [inline ingestion/utils.py:11]: What's `st` stand for?

File: pyproject.toml [hunk 3]
```diff
@@ -12,7 +12,7 @@ boto3 = "^1.34.119"
 datahub = "^0.999.1"
 moto = "4"
 pandas = "^2.2.2"
-acryl-datahub = { extras = ["dbt"], version = "~=0.13.3.4" }
+acryl-datahub = { extras = ["dbt"], version = "==0.13.2.4" }
```
Comment (MatMoore) [inline pyproject.toml:15]: is the downgrading here intentional?

File: tests/utils.py [hunk 4]
```diff
@@ -6,7 +6,7 @@
 from datahub.ingestion.api.common import EndOfStream, PipelineContext, RecordEnvelope
 from datahub.ingestion.transformer.dataset_transformer import DatasetTransformer
 from datahub.metadata.schema_classes import MetadataChangeEventClass
-from datahub.utilities.urns.urn import Urn
+from datahub.utilities.urns._urn_base import Urn
```
Comment (MatMoore) [inline tests/utils.py:9]: why are you importing from this private module? What's the difference between this `Urn` and `urns.urn.Urn`?

**Review by MatMoore — COMMENTED** [review]
Looks good overall, I'm just a bit confused about a few bits - see inline comments

File: pyproject.toml (hunk 3 above)
Comment (tom-webber) [reply pyproject.toml:15]: This should probably be reverted and applied via a different PR. The `0.13.3.X` ingestions are [failing with an ingestion error on `main`](https://mojdt.slack.com/archives/C03QZ776JVA/p1721747684834469?thread_ts=1721745324.616259&cid=C03QZ776JVA)

File: tests/utils.py (hunk 4 above)
Comment (tom-webber) [reply tests/utils.py:9]: This fixes a type error for me (via `"python.analysis.typeCheckingMode": "basic"` in vscode)
> "Urn" is not exported from module "datahub.utilities.urns.urn"
  Import from "datahub.utilities.urns._urn_base" insteadPylance[[reportPrivateImportUsage]](https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportPrivateImportUsage)

File: ingestion/utils.py (hunk 2 above)
Comment (tom-webber) [reply ingestion/utils.py:11]: No idea - modelled after something from SO. Can be changed to something more meaningful.

File: ingestion/transformers/assign_cadet_databases.py (hunk 1 above)
Comment (tom-webber) [reply ingestion/transformers/assign_cadet_databases.py:23]: > Was the intention to set a logger variable and use it?

This

File: ingestion/utils.py (hunk 2 above)
Comment (MatMoore) [reply ingestion/utils.py:11]: maybe `report_time`?

File: pyproject.toml (hunk 3 above)
Comment (MatMoore) [reply pyproject.toml:15]: Yeah I think a separate PR would be good, makes it easier to roll back either part if we need to.

Do we have a ticket to upgrade again and fixt how we create the Domain entities?

**Comment by MatMoore** [comment]
This is working on dev, and output confirms our hypothesis that `AssignCadetDomains` is what is slowing things down. Pretty sure we can significantly optimise this!

- "create cadet domains" 2m 48s
	- `__init__` 0s
	- `get_workunits` 2m 37s
		- `get_cadet_mainfest` 2m 37s
		- `_get_databases_with_domains_and_display_tags` 0s
- "push metadata to datahub" 3h 11m 32s
	- `__init__` 0s
	- `get_workunits` 2m 27s
		- `_get_cadet_manifest` 2m 36s
		- `_get_domain_mapping` 0s
	- `AssignCadetDomains` transformer 2h 32m 10s
	- AssignCadetDatabases 2m 42s
		- `get_cadet_manifest` 2m 41s
		- `_get_table_database_mappings` 0s

**Comment by MatMoore** [comment]
@hjribeiro-moj would you mind rereviewing? I had to push up a small fix yesterday

**Review by mitchdawson1982 — APPROVED** [review]
LGTM

**Pull request merged**